pyccel/parser/semantic.py

```diff
@@ -3829,6 +3829,15 @@ def _visit_Assign(self, expr):
                     new_expressions.append(Assign(rhs_var, rhs))
                     rhs = rhs_var
                 lhs = PythonTuple(*new_lhs)
+            elif isinstance(rhs, PyccelFunction):
+                assert isinstance(rhs.class_type, InhomogeneousTupleType)
+                r_iter = [self.scope.collect_tuple_element(v) for v in rhs]
+                new_lhs = []
+                for i,(l,r) in enumerate(zip(lhs, r_iter)):
+                    d = self._infer_type(r)
+                    new_lhs.append( self._assign_lhs_variable(l, d, r, new_expressions,
+                                                    arr_in_multirets=r.rank>0 ) )
+                lhs = PythonTuple(*new_lhs)
```

Comment:
What is going on here? And how is it related to this PR?

Reply:
Fascinating! Does this allow to handle something else (on top of `popitem`) which was not handled before?

Reply:
Nothing comes to mind but it should be relevant for any built-in that returns an inhomogeneous tuple